### ISSUE TITLE
Cannot load onnx model

### ISSUE THREAD

**Issue by ThiagoMateo** [NVIDIA/tensorrt-inference-server#911]
I load my onnx model without condig.pbtxt file, but i got error: Mismatch between allocated memory size

```
trtserver: engine.cpp:1094: bool nvinfer1::rt::Engine::deserialize(const void*, std::size_t, nvinfer1::IGpuAllocator&, nvinfer1::IPluginFactory*): Assertion `size >= bsize && "Mismatch between allocated memory size and expected size of serialized engine."' failed.
```

**Comment by CoderHam** [comment]
This error seems to be caused by TRTIS thinking it's a TRT model. Can you verify it is indeed an onnx model?

**Comment by ThiagoMateo** [comment]
how to verify my model is onnx?
i also try to run with config.pbtxt file
```
name: "faces"
platform: "onnxruntime_onnx"
max_batch_size: 4
input [
  {
    name: "data"
    data_type: TYPE_FP32
    format: FORMAT_NCHW
    dims: [ 3, 112, 112 ]
  }
]
output [
  {
    name: "fc1"
    data_type: TYPE_FP32
    dims: [ 1, 512 ]
  }
]
instance_group [
  {
    kind: KIND_CPU
    count: 1
  }
]

```
but got another error:
```
unexpected platform type onnxruntime_onnx for faces
``

**Comment by ThiagoMateo** [comment]
The cause of this error is that I use version 19.03.
Thank for your help. But i got  another error:
```
I1126 09:46:30.040848 1 onnx_backend.cc:185] Creating instance faces_0_0_cpu on CPU using model.onnx
2019-11-26 09:46:30.258737129 [W:onnxruntime:log, graph.cc:2227 CleanUnusedInitializers] pre_fc1_bia
 exists in this graph's initializers but it is not used by any node
2019-11-26 09:46:30.384838746 [E:onnxruntime:InferenceSession, inference_session.cc:529 Initialize] 
xception during initialization: /workspace/onnxruntime/onnxruntime/core/providers/cpu/nn/batch_norm.
:39 onnxruntime::BatchNorm<T>::BatchNorm(const onnxruntime::OpKernelInfo&) [with T = float] spatial 
= 1 was false. BatchNormalization kernel for CPU provider does not support non-spatial cases
Stacktrace:

E1126 09:46:30.406133 1 model_repository_manager.cc:779] failed to load 'faces' version 1: Internal:
onnx runtime error 6: Exception during initialization: /workspace/onnxruntime/onnxruntime/core/provi
ers/cpu/nn/batch_norm.h:39 onnxruntime::BatchNorm<T>::BatchNorm(const onnxruntime::OpKernelInfo&) [w
th T = float] spatial == 1 was false. BatchNormalization kernel for CPU provider does not support no
-spatial cases
Stacktrace:

```
Can i use trtis to deploy my model?

**Comment by GuanLuo** [comment]
@ThiagoMateo From ONNX Runtime's commit history, it appears that non-spatial BatchNormalization is supported since ONNX Runtime v1.0.0. TRTIS just advance the ONNX Runtime version to 1.0.0 recently, which will be in 19.11. So you can wait until 19.11 is released, in the next few days, and see if TRTIS can load the model successfully.

At the meantime, you may try to deploy your model on GPU / with different [Execution Accelerators](https://docs.nvidia.com/deeplearning/sdk/tensorrt-inference-server-master-branch-guide/docs/protobuf_api/model_config.proto.html#_CPPv4N6nvidia15inferenceserver23ModelOptimizationPolicy21ExecutionAcceleratorsE) as the error is just indicating that the CPU provider doesn't support this op (while the others may support).

**Comment by ThiagoMateo** [comment]
@GuanLuo  i use trtis 19.11. Trtis cannot load my model but it doesn't show me the error.
```
NVIDIA Release 19.11 (build 8810908)

Copyright (c) 2018-2019, NVIDIA CORPORATION.  All rights reserved.

Various files include modifications (c) NVIDIA CORPORATION.  All rights reserved.
NVIDIA modifications are covered by the license terms that apply to the underlying
project or file.

NOTE: The SHMEM allocation limit is set to the default of 64MB.  This may be
   insufficient for the inference server.  NVIDIA recommends the use of the following flags:
   nvidia-docker run --shm-size=1g --ulimit memlock=-1 --ulimit stack=67108864 ...

2019-11-27 13:42:20.545350: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.2
I1127 13:42:20.601008 1 metrics.cc:160] found 1 GPUs supporting NVML metrics
I1127 13:42:20.606535 1 metrics.cc:169]   GPU 0: GeForce GTX 1070
I1127 13:42:20.606883 1 server.cc:112] Initializing TensorRT Inference Server
2019-11-27 13:42:21.680960719 [W:onnxruntime:log, graph.cc:2367 CleanUnusedInitializers] Removing initializer 'pre_fc1_bias'. It is not used by any node and should be removed from the model.
E1127 13:42:21.935151 1 model_repository_manager.cc:1453] model input NHWC/NCHW require 3 dims for faces
E1127 13:42:21.935289 1 main.cc:1097] error: creating server: INTERNAL - failed to load all models

```

my config file:
```
input [
  {
    name: "data"
    data_type: TYPE_FP32
    format: FORMAT_NCHW
    dims: [ 1, 3, 112, 112 ]
  }
]
output [
  {
    name: "fc1"
    data_type: TYPE_FP32
    dims: [ 1, 512 ]
  }
]
instance_group [
  {
    kind: KIND_CPU
    count: 1
  }
]

```

**Comment by GuanLuo** [comment]
@ThiagoMateo if you specifies `format` field in the input, the input should only has 3 dimensions specified (c, h, w). You can try removing the `format` field.
